pkg/scheduler/core/generic_scheduler.go

```diff
@@ -214,6 +214,21 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 		return result, err
 	}
 
+	nodeToStatuses := make(framework.NodeToStatusesMap, len(failedPredicateMap))
+	for nodeName, reasons := range failedPredicateMap {
```

Comment:
I think we want findNodesThatFit to return NodeToStatusesMap type (call it filteredNodesStatuses) in addition to failedPredicateMap. This means couple of things:

1) add a NodeToStatusesMap member (filteredNodesStatuses) to [FitError](https://github.com/kubernetes/kubernetes/blob/c445929c18534b86ad8ce9b90ee5aee24319a5c3/pkg/scheduler/core/generic_scheduler.go#L90).

2) iterates over the filteredNodesStatuses in [Error](https://github.com/kubernetes/kubernetes/blob/c445929c18534b86ad8ce9b90ee5aee24319a5c3/pkg/scheduler/core/generic_scheduler.go#L107) after the failedPredicateMap iteration.

As part of fixing #80775, we will add logic to nodesWherePreemptionMightHelp to examine filteredNodesStatuses for filters that can help with preemption.

The idea is to really separate predicate reason from filter statues. Moreover, I don't think we want to pass **predicate** failures to post-filter, I think we only want to pass **filters** failures only, and so separating the maps will help with that as well.

Reply:
> Moreover, I don't think we want to pass **predicate** failures to post-filter, I think we only want to pass **filters** failures only

I agree on this, WDYT @bsalamat

Reply:
+1. Thanks!